### ISSUE TITLE
loop error rules writing a function

### ISSUE THREAD

**Issue by b1tw0rker** [typescript-eslint/typescript-eslint#5670]
### Before You File a Bug Report Please Confirm You Have Done The Following...

- [X] I have tried restarting my IDE and the issue persists.
- [X] I have updated to the latest version of the packages.
- [X] I have [searched for related issues](https://github.com/typescript-eslint/typescript-eslint/issues?q=is%3Aissue+label%3Abug+label%3A%22package%3A+eslint-plugin%22) and found none that matched my issue.
- [X] I have [read the FAQ](https://typescript-eslint.io/docs/linting/troubleshooting) and my problem is not listed.

### Playground Link

https://github.com/prettier/eslint-plugin-prettier#options

### Repro Code

```typescript
Rule
https://eslint.org/docs/latest/rules/space-before-function-paren
shades 
prettier rule  https://github.com/prettier/eslint-plugin-prettier#options

this produces a error loop.
The one rule want a space the other rules do not want a space...

In my opinion eslint does just one thing: "Slows down development" :-(
```

### ESLint Config

```javascript
much config
```

### tsconfig

```jsonc
{
  "compilerOptions": {
    // ...
  }
}
```

### Expected Result

Make a decision which rule should be work fine. I guess, there are 1000 eslint rules which could be deleted....

### Actual Result

Both ruselt shades each other.

### Versions

| package                                | version |
| -------------------------------------- | ------- |
| `@typescript-eslint/eslint-plugin`     | `X.Y.Z` |
| `@typescript-eslint/parser`            | `X.Y.Z` |
| `TypeScript`                           | `X.Y.Z` |
| `ESLint`                               | `X.Y.Z` |
| `node`                                 | `X.Y.Z` |

**Comment by bradzacher** [comment]
I'm not sure why you filed this rather vague issue in our repo when you're clearly unhappy with prettier.

Regardless - you linked your solution. Just scroll up just a little bit in the eslint-plugin-prettier readme.
https://github.com/prettier/eslint-plugin-prettier#recommended-configuration
